src/test/scala/edu/ie3/simona/model/participant/storage/StorageMathFlexModelSpec.scala

```diff
@@ -0,0 +1,415 @@
+/*
+ * © 2025. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.participant.storage
+
+import edu.ie3.simona.model.em.OptimizedFlexStrat
+import edu.ie3.simona.model.participant.storage.StorageMathFlexModel.StorageMathFlexOptions
+import edu.ie3.simona.model.participant.storage.StorageMathFlexModelSpec.*
+import edu.ie3.simona.test.common.{MathFlexTestLike, UnitSpec}
+import edu.ie3.util.scala.quantities.DefaultQuantities.{zeroKW, zeroKWh}
+import optimus.algebra.{Double2Const, Zero}
+import optimus.optimization.*
+import optimus.optimization.enums.{SolutionStatus, SolverLib}
+import optimus.optimization.model.MPFloatVar
+import org.scalatest.OptionValues
+import squants.energy.{Energy, KilowattHours, Kilowatts, WattHours}
+import squants.time.Hours
+import squants.{Dimensionless, Each, Power, Time}
+
+import java.util.UUID
+
+class StorageMathFlexModelSpec extends UnitSpec with MathFlexTestLike {
+
+  // Testing tolerances
+  given Double = 1e-6
+  given Energy = WattHours(1e-9)
+  val constraintTolerance = 1e-3
+
+  "StorageMathFlexModelSpec" should {
+
+    "be created with correct parameters" in {
+
+      val currentEnergy = KilowattHours(10)
+      val eStorage = KilowattHours(20)
+      val pMax = Kilowatts(10)
+
+      val etas = Seq(.6, .65, .7, .75, .8, .85, 0.9, .92, .95, .98, 1)
+
+      forEvery(Table("etaCharging", etas*)) { etaCharging =>
+        forEvery(Table("etaDischarging", etas*)) { etaDischarging =>
+
+          val classic = ClassicModel(
+            currentEnergy = currentEnergy,
+            eStorage = eStorage,
+            pMax = pMax,
+            etaCharging = Each(etaCharging),
+            etaDischarging = Each(etaDischarging),
+          )
+
+          // charging until full, with maximum power
+          val power1 = pMax
+          val duration1 = (eStorage - currentEnergy) / (power1 * etaCharging)
+          // discharging three quarters, with half power
+          val power2 = -pMax / 2
+          val time2 = eStorage * 0.75 / (-power2 * 1 / etaDischarging)
+          // charging until half, with quarter power
+          val power3 = pMax / 4
+          val time3 = eStorage / 4 / (power3 * etaCharging)
```

Comment:
```suggestion
          val time3 = eStorage / 2 / (power3 * etaCharging)
```

or "charging until one quarter" in the comment above, or?

Reply:
No, because we charged to full, then discharged three quarters, thus we're at one quarter and need one more quarter to reach half full.

Reply:
Makes sense, didn't get that it these steps are seen as a sequence.